### PR TITLE
CW-203 Menu redesign

### PR DESCRIPTION
**Pull request by sergiomercado19**
# Change

- _Brought forward functionality from the old Menu design._
- _Implemented high fidelity Menu wireframe using typography._

## Change reason

This refactoring was needed as we begin implementing all high-fidelity wireframes in preparation for our showcase.

## Comments

I have used the spacing variables defined in `_spacing.scss` . Make sure that the spacing is as envisioned. For the spacing between the links (both page and social) I have used the padding property by adding half the desired space to each link.

### CODE REVIEW

File: frontend/src/components/Menu.vue [hunk 1]
```diff
@@ -6,23 +6,70 @@
...
+            </a>
+          </div>
+        </div>
+      </div>
+      <!-- Footer -->
+      <div class="spacing-xs" />
+      <div>
+        <p style="display: inline;">
+          <img height="14px" src="@/assets/moon-icon.png" style="display: inline; padding-right: 5px;">
+          Only Dark mode allowed here at CSESoc. Our apologies.
```
Comment (glebme) [inline frontend/src/components/Menu.vue:69]: This isn't displayed.

File: frontend/src/components/Menu.vue [hunk 2]
```diff
@@ -6,23 +6,70 @@
...
-      </RouterLink>
-      <RouterLink to="/engage" class="link">
-        <label>Engage</label>
+  <div @click.stop="hide" class="cover">
+    <div class="frame">
+      <!-- Logo -->
+      <RouterLink to="/" v-on:click.native="showMenu = false">
+        <div class="fill-height" style="width: 140px; margin-left: 5px;">
+          <v-img src="@/assets/csesoc-logo-white.svg" />
+        </div>
```
Comment (glebme) [inline frontend/src/components/Menu.vue:15]: This is not displayed.

File: frontend/src/components/Menu.vue [hunk 3]
```diff
@@ -6,23 +6,70 @@
...
+            <a class="link" href="https://www.youtube.com/channel/UC1JHpRrf9j5IKluzXhprUJg">
+              <p>Youtube</p>
+            </a>
+            <a class="link" href="https://www.linkedin.com/company/csesoc/">
+              <p>LinkedIn</p>
+            </a>
+            <a class="link" href="#">
+              <p>TikTok</p>
+            </a>
+          </div>
```
Comment (glebme) [inline frontend/src/components/Menu.vue:61]: _nit_: spacing here is a bit wide for me

File: frontend/src/components/Menu.vue [hunk 4]
```diff
@@ -39,47 +86,108 @@ export default {
...
+    padding: $space-xs/2; /* Add some padding */
+    padding-left: 0px; /* Keep it left aligned */
+    cursor: pointer; /* To simulate a link */
+    font-size: $text-md;
+    font-weight: normal;
+    transition: all .2s ease-in-out;
+
+    h2 {
+      cursor: pointer; /* To simulate a link */
+    }
```
Comment (glebme) [inline frontend/src/components/Menu.vue:140]: _nit_: We need to standardise CSS/SCSS comments in our style guide.

**Review by glebme — CHANGES_REQUESTED** [review]
Design: LGTM
Functionality: No utilisation of the grid column system leading to the overlapping containers as windows are resized. Would you think implementing the grid system that we have would help this.
Complexity: LGTM
Tests: Not accompanied with a test. 
Naming: spacing-md, spacing-xs may confuse developers that think its the same as the encoded values for these variables in the SCSS.
Comments: LGTM
Style: Problem -> Need to write one up.
Documentation: We need to begin writing documentation up to generalise styles. Another thing to add to our list of things to do.
Overall I believe this gives credit to the design of the high fidelity wireframe. What I am concerned now is for future developers and the definite need to draw up concrete living frontend style guide to inform developers on how to go about create components. Once Jarvis reviews on the design side can this PR be closed and the creation of a design implementation guideline be drawn up to progress.
Screen Shot 2020-08-27 at 3 04 26 pm

File: frontend/src/components/Menu.vue (hunk 3 above)
Comment (sergiomercado19) [reply frontend/src/components/Menu.vue:61]: This is a SCSS issue, will get fixed on another PR.

File: frontend/src/components/Menu.vue (hunk 1 above)
Comment (sergiomercado19) [reply frontend/src/components/Menu.vue:69]: I can see this in the screenshot, it's just the colour that is very hard to see.

File: frontend/src/components/Menu.vue (hunk 2 above)
Comment (sergiomercado19) [reply frontend/src/components/Menu.vue:15]: This logo should be visible 🤔 it's imported the same way as the main logo on our navbar. Maybe take make sure that the SVG is actually in the assets folder.

**Review by JarvisWang0715 — CHANGES_REQUESTED** [review]
The font size of the content on the left seems a bit off to me. It is supposed to be h2 (30 pt) while it seems like it is h1 (60pt).

**Comment by sergiomercado19** [comment]
> * The font size of the content on the left seems a bit off to me. It is supposed to be h2 (30 pt) while it seems like it is h1 (60pt).

Fixed. It was getting overridden, my bad.

**Comment by sergiomercado19** [comment]
The left and right margins are larger and change dynamically because I'm using `v-containers` as these are required for Vuetify's grid system.
This is how the menu looks for me:
image

**Review by JarvisWang0715 — COMMENTED** [review]
The text-size value of of the page-link: hover should be $text-lg

**Comment by sergiomercado19** [comment]
> The text-size value of of the page-link: hover should be $text-lg

@JarvisWang0715 Fixed. Can you please approve if it looks good to you.

**Review by JarvisWang0715 — APPROVED** [review]

**Review by glebme — APPROVED** [review]
Those images aren't loading because of the browser. This is an issue for another PR. Pretty sure it has something to do with the svg.

Everything I have left is a _nit_, so when free and time appropriate if a pull request (if needed) could be open to address this
- to perhaps improve the jitter when hovering over the links maybe look at reducing the size of the <a> to only be size of the text. At the moment it flexes to meet the container.
- Breakpoints, e.g. resizing past a certain point stops the dynamic resizing. Dont know how we go about addressing this
- With the footer because row was used to wrap content, `row` instead of `div` should be used to wrap the bottom part too.

**Pull request merged**